metrics/opentelemetry.go

```diff
@@ -8,38 +8,48 @@ import (
 
 // Measures
 var (
-	BitswapRequestCount                      = stats.Int64("bitswap_request_total", "The number of bitswap requests received", stats.UnitDimensionless)
-	BitswapResponseCount                     = stats.Int64("bitswap_response_total", "The number of bitswap responses", stats.UnitDimensionless)
-	BitswapRetrieverRequestCount             = stats.Int64("bitswap_retriever_request_total", "The number of bitswap messages that required a retriever lookup", stats.UnitDimensionless)
-	BlockstoreCacheHitCount                  = stats.Int64("blockstore_cache_hit_total", "The number of blocks from the local blockstore served to peers", stats.UnitDimensionless)
-	BytesTransferredTotal                    = stats.Int64("data_transferred_bytes_total", "The number of bytes transferred from storage providers to retrieval clients", stats.UnitBytes)
-	BitswapRequestWithIndexerCandidatesCount = stats.Int64("bitswap_request_with_indexer_candidates_total", "The number of bitswap requests that result in non-zero candidates from the indexer", stats.UnitDimensionless)
-	BitswapRequestWithSuccessfulQueryCount   = stats.Int64("bitswap_request_with_successful_query_total", "The number of bitswap requests that result in non-zero number of successful queries from SPs", stats.UnitDimensionless)
```

Comment:
Renaming these is going to cause a data migration hiccup in our stats for anyone doing historical queries; or we're going to have to do some `old+new` stuff in there; that's a tiny bit annoying? I appreciate the cleanup here, these names were my fault but they seemed reasonable as an extension of "request" coming out of "bitswap".

I'll leave this call to you but can you look at the query in the pipeline bar graph on our dashboard and consider how tricky it might be to make it so it can include new+old and not just jump to zero if someone does a range that spans back beyond the time we deploy this.

Reply:
This is true, and I did consider the consequences of this change. I have a couple thoughts on this.

If I could have it 100% my way, I care more about the maintainability of the code, but I'm obviously biased. :P I already find the metric names a little unruly, so keeping them consistent and apparently relatable to one another makes it easier to maintain. I would also argue changing the actual metric names makes searching through Grafana for specific metrics easier, making it easier to construct new views on the dashboard. As much that I do agree that changing naming references like this is probably not the best for historic viewing, the best time to fix naming of things like this is yesterday. This particular dashboard panel is still relatively new, so it'd probably be easier to get away with it now than later. I do feel this would become less of a problem as time goes on.

However, I realize there are other stakeholders here that might find the data skip confusing when looking back far enough. We could definitely do a `old+new` dashboard calculation in the meantime though as a compromise. Got any opinions @hannahhoward?

I see a few paths forward, in order of effort required to keep all stakeholders happy.
1) Completely revert these changes
2) Keep the changes made to the metric variables but revert the metric names back to `bitswap_*`
3) Keep all the changes, deal with data skip in historic views with updated dashboard metrics.

Reply:
> This particular dashboard panel is still relatively new

yep, very new, let's just go with it and cope as we go